metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -956,7 +974,7 @@ ControllerResult<AlterPartitionResponseData> alterPartition(AlterPartitionReques
                     partitionId,
                     clusterControl::active,
                     featureControl.metadataVersion().isLeaderRecoverySupported());
-                if (configurationControl.uncleanLeaderElectionEnabledForTopic(topicData.name())) {
+                if (configurationControl.uncleanLeaderElectionEnabledForTopic(topicData.topicName())) {
```

Comment:
I might have missed it, but do we set the topic name somewhere if the request was passed the TopicId instead?